solidity/contracts/isms/hook/PolygonZkevmIsm.sol

```diff
@@ -0,0 +1,178 @@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {AbstractCcipReadIsm} from "../ccip-read/AbstractCcipReadIsm.sol";
+import {IMailbox} from "../../interfaces/IMailbox.sol";
+import {IPolygonZkEVMBridge} from "../../interfaces/polygonzkevm/IPolygonZkEVMBridge.sol";
+import {AbstractMessageIdAuthorizedIsm} from "../hook/AbstractMessageIdAuthorizedIsm.sol";
+import {ICcipReadIsm} from "../../interfaces/isms/ICcipReadIsm.sol";
+import {LibBit} from "../../libs/LibBit.sol";
+import {IBridgeMessageReceiver} from "../../interfaces/polygonzkevm/IBridgeMessageReceiver.sol";
+
+// ============ External Imports ============
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title PolygonZkevmIsm
+ * @notice Polygon zkEVM chain Ism that uses the Polygon zkEVM bridge to verify messages
+ */
+contract PolygonZkevmIsm is
+    ICcipReadIsm,
+    AbstractMessageIdAuthorizedIsm,
+    IBridgeMessageReceiver
+{
+    using Message for bytes;
+    using LibBit for uint256;
+    using TypeCasts for bytes32;
+
+    IMailbox public mailbox;
+    string[] public offchainUrls;
+
+    // ============ Constants ============
+    IPolygonZkEVMBridge public immutable zkEvmBridge;
+    uint8 public constant override moduleType =
+        uint8(IInterchainSecurityModule.Types.CCIP_READ);
+
+    // ============ Constructor ============
+    constructor(
+        address _zkEvmBridge,
+        address _mailbox,
+        string[] memory _offchainUrls
+    ) {
+        require(
+            Address.isContract(_zkEvmBridge),
+            "PolygonZkevmIsm: invalid ZkEVMBridge"
+        );
+        require(
+            Address.isContract(_zkEvmBridge),
+            "PolygonZkevmIsm: invalid IMailbox"
+        );
+        zkEvmBridge = IPolygonZkEVMBridge(_zkEvmBridge);
+        mailbox = IMailbox(_mailbox);
+        offchainUrls = _offchainUrls;
+    }
+
+    /**
+     * @dev off-chain verification information for a given message.
+     * @param _message The message for which off-chain verification information is requested.
+     */
+    function getOffchainVerifyInfo(
+        bytes calldata _message
+    ) external view override {
+        bytes memory messageId = abi.encodePacked(_message.id());
+        revert OffchainLookup(
+            address(this),
+            offchainUrls,
+            messageId,
+            PolygonZkevmIsm.verify.selector,
+            _message
+        );
+    }
+
+    /**
+     * @dev Calls the Polygon zkEVM bridge to claim the message.
+     * @param _metadata from CCIP call
+     * @return A boolean indicating whether the message was successfully verified and processed.
+     */
+    function verify(
+        bytes calldata _metadata,
+        bytes calldata
```

Comment:
```suggestion
        bytes calldata message
```
you need to compare `message.id()` to the `payload` you passed on the origin chain